src/node_contextify.h

```diff
@@ -0,0 +1,100 @@
+#include "node_internals.h"
+#include "node_watchdog.h"
+#include "base_object-inl.h"
+
+namespace node {
+
+using v8::Array;
+using v8::Boolean;
+using v8::Context;
+using v8::FunctionCallbackInfo;
+using v8::Local;
+using v8::Name;
+using v8::Object;
+using v8::Persistent;
+using v8::PropertyCallbackInfo;
+using v8::PropertyDescriptor;
+using v8::Value;
+using v8::WeakCallbackInfo;
+
+class ContextifyContext {
+ protected:
+  // V8 reserves the first field in context objects for the debugger. We use the
+  // second field to hold a reference to the sandbox object.
+  enum { kSandboxObjectIndex = 1 };
+
+  Environment* const env_;
+  Persistent<Context> context_;
+
+ public:
+  ContextifyContext(Environment* env, Local<Object> sandbox_obj);
+  ~ContextifyContext();
+
+  Local<Value> CreateDataWrapper(Environment* env);
+  Local<Context> CreateV8Context(Environment* env, Local<Object> sandbox_obj);
+  static void Init(Environment* env, Local<Object> target);
+  static void MakeContext(const FunctionCallbackInfo<Value>& args);
+  static void IsContext(const FunctionCallbackInfo<Value>& args);
+  static void WeakCallback(const WeakCallbackInfo<ContextifyContext>& data);
+  static ContextifyContext* ContextFromContextifiedSandbox(
```

Comment:
Do these functions need to be public? Please try making private as many functions as possible. (Especially the interceptor callback functions.)